### ISSUE TITLE
rpkid scheduler needs work

### ISSUE THREAD

**Issue by sraustein** [dragonresearch/rpki.net#291]
There are a couple of issues with the current rpkid scheduler.

The core loop is fairly aggressive about prioritizing deferred events over running the I/O or timer subsystems.   Reason for this is lost in the mists of time but probably was a correction for an earlier problem where it wasn't aggressive enough.   May need to revisit this, but assume for the moment that we're keeping the core loop as it stands.

This explains why my experiment yesterday didn't help: just using the async loop system isn't enough to yield control back to the I/O system, for that we need to use something like a timer with a one or two second delay.   Might want to add a utility function event_yield() which does that.  Given such a utility routine we could probably (have to test) solve at least part of #274 just by using event_yield(iterator) at the end of the ROA construction loop occasionally, say, after constructing 50 new ROAs.   This wouldn't mess things up the way that dismissing from the ROA construction loop entirely does, but would still allow I/O to run, and would be a relatively minor code change.

As a separate issue, I'm not happy with the current business of running the cron code both from the rpkid-wide internal cron timer and also in response to "run_now" events signaled via left-right protocol.  When things get slow enough, this can mean we have multiple instances of the same loop running in parallel on the same <self/>, which seems risky even if the right thing usually happens.   This almost certainly needs some kind of restructuring, perhaps including some hack where we keep track of how recently we ran .cron() for each <self/> (and whether we're running one now) and use that to make sure we don't run .cron() more than once for a single <self/> at a time.  Dunno if any of that data needs to go in SQL.

_Trac ticket #275 component rpkid priority blocker, owner sra, created by sra on 2012-07-31T14:22:38Z, last modified 2012-10-09T23:28:29Z_
